# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientInfo.cpp
# Commit: ddf4012a7652
# Full Hash: ddf4012a7652e36d144fc43bc99335276deeafbe
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-05-22 03:40:58
# Regressor Bug: 965637
# File Overlap Count: 1
# Description:
#   Bug 965637: Move CSP from Principal into Client, part 1: backend changes. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27654
# ==============================================================================

diff -r 6c2047bc2f3e -r ddf4012a7652 dom/clients/manager/ClientInfo.cpp
--- a/dom/clients/manager/ClientInfo.cpp	Tue May 21 22:51:54 2019 +0000
+++ b/dom/clients/manager/ClientInfo.cpp	Tue May 21 23:14:27 2019 +0000
@@ -20,7 +20,9 @@
                        const TimeStamp& aCreationTime)
     : mData(MakeUnique<IPCClientInfo>(aId, aType, aPrincipalInfo, aCreationTime,
                                       EmptyCString(),
-                                      mozilla::dom::FrameType::None)) {}
+                                      mozilla::dom::FrameType::None,
+                                      mozilla::Nothing(), mozilla::Nothing())) {
+}
 
 ClientInfo::ClientInfo(const IPCClientInfo& aData)
     : mData(MakeUnique<IPCClientInfo>(aData)) {}
@@ -97,5 +99,22 @@
   return ref;
 }
 
+const Maybe<mozilla::ipc::CSPInfo>& ClientInfo::GetCspInfo() const {
+  return mData->cspInfo();
+}
+
+void ClientInfo::SetCspInfo(const mozilla::ipc::CSPInfo& aCSPInfo) {
+  mData->cspInfo() = Some(aCSPInfo);
+}
+
+const Maybe<mozilla::ipc::CSPInfo>& ClientInfo::GetPreloadCspInfo() const {
+  return mData->preloadCspInfo();
+}
+
+void ClientInfo::SetPreloadCspInfo(
+    const mozilla::ipc::CSPInfo& aPreloadCSPInfo) {
+  mData->preloadCspInfo() = Some(aPreloadCSPInfo);
+}
+
 }  // namespace dom
 }  // namespace mozilla